src/net/udp/mod.rs

```diff
@@ -415,15 +417,19 @@ impl<T> Future for RecvDgram<T>
     type Error = io::Error;
 
     fn poll(&mut self) -> Poll<Self::Item, io::Error> {
-        let (n, addr) = {
-            let (ref socket, ref mut buf) =
-                *self.0.as_mut().expect("RecvDgram polled after completion");
-
-            try_nb!(socket.recv_from(buf.as_mut()))
+        let (sock, mut buf) = match self.0.take() {
```

Comment:
Similar to the function above, could this be left as is (avoid extraneous moves in the prior version)